src/lib.rs

```diff
@@ -67,10 +68,33 @@ pub struct RenderPass {
     texture_version: Option<u64>,
     next_user_texture_id: u64,
     pending_user_textures: Vec<(u64, egui::Texture)>,
-    user_textures: Vec<Option<wgpu::BindGroup>>,
+    //for support changing texture
+    user_textures: Vec<Option<(Arc<wgpu::Texture>,wgpu::BindGroup)>>,
 }
 
 impl RenderPass {
+    /// Assume wgpu::Texture as egui::TextureId
+    ///  i.e given wgpu::Texture (wgpu Texture handle) as egui::TextureId (egui texture handle)without allocation
+    ///  this method return  only egui::TextureId::UserTexture(u64)
+    pub fn texture_as_egui_texture_id(&mut self,device:&wgpu::Device ,texture_handle:Arc<wgpu::Texture>)->egui::TextureId{
```

Comment:
A run with cargo fmt would also be nice.

Reply:
ok i will try it